app/javascript/controllers/area_served_controller.js

```diff
@@ -0,0 +1,29 @@
+import { Controller } from "@hotwired/stimulus"
+export default class extends Controller {
+    static targets= ["share", "total", "warning"]
+    connect () {
+        this.calculateClientShareTotal()
+    }
+    calculateClientShareTotal(){
+        let total = 0;
+        let share_targets = this.shareTargets
+        share_targets.forEach(
+            share_target =>{
+                if(share_target.value){
+                    total += parseInt(share_target.value);
+                }
+            }
+        )
+
+        this.totalTarget.innerHTML = total + " %"
+
+        if(total == 0 || total == 100){
+            this.warningTarget.style.visibility= 'hidden';
+        }else {
+            this.warningTarget.style.visibility= 'visible';
+            this.warningTarget.style.color= 'red';
+        }
+        return total;
+    }
```

Comment:
In general there are quite a lot of spacing fixes I'd put here (we don't have a JS linter):

*   A space before and after =>
*   A space before and after { and } 
*   A space before ( and after )
*   A space before and after =

Reply:
Still missing some spaces before `=` :smile: